src/mrinufft/io/utils.py

```diff
@@ -0,0 +1,38 @@
+from mrinufft.operators.base import with_numpy_cupy, get_array_module
+
+
+@with_numpy_cupy
+def add_phase_to_kspace_with_shifts(kspace_data, kspace_loc, normalized_shifts):
```

Comment:
`with_numpy_cupy` expect a signature with (self, data, output=None). I think you are going to get trouble here.